src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Unix.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Internal.Runtime.Augments
+{
+    public sealed partial class RuntimeThread
+    {
+        private readonly WaitSubsystem.ThreadWaitInfo _waitInfo;
+
+        internal WaitSubsystem.ThreadWaitInfo WaitInfo => _waitInfo;
+
+        public ApartmentState GetApartmentState()
+        {
+            Debug.Fail("Should not reach here because CoreFX's Thread class does not forward to this function on Unix");
+            return ApartmentState.Unknown;
```

Comment:
Should these functions just throw then if we don't expect them to ever be called? e.g. ```throw null``` like we have in the refs?